tests/store/tracking/test_file_store.py

```diff
@@ -157,6 +158,31 @@ def test_valid_root(self):
         with pytest.raises(Exception, match=r"does not exist"):
             second_file_store._check_root_dir()
 
+    def test_removing_default_experiment_recreates_default_id(self):
+        def _is_default_in_experiments(view_type):
+            search_result = file_store.search_experiments(view_type=view_type)
+            ids = [experiment.experiment_id for experiment in search_result]
+            return FileStore.DEFAULT_EXPERIMENT_ID in ids
+
+        file_store = FileStore(self.test_root)
+        self.assertTrue(_is_default_in_experiments(ViewType.ACTIVE_ONLY))
+        # Soft-delete the default experiment
+        file_store.delete_experiment(FileStore.DEFAULT_EXPERIMENT_ID)
```

Comment:
Didn't know the default experiment can be deleted.

Reply:
I don't think it's recommended to. Do you think it's pointless to test this?

Reply:
I think we should not allow users to delete the default experiment.

Reply:
I made some adjustments to the logic to avoid the identity insert logic that was there before. Instead of regenerating the default experiment_id and using that for the current `create_experiment()` call, I'm issuing a warning that recreates the default experiment and then continuing generating a new experiment with a two-row commit within `create_experiment()` so that the tracking server doesn't end up in a non-functional state if someone dropped a row entry or deleted the local fs directory for experiment_id 0. This will be largely silent (with just a warning emitted) but no change to the expected behavior of create_experiment(). 
I added some validation steps to the unit test for this.